packages/contracts-bedrock/src/L2/CrossL2Inbox.sol

```diff
@@ -30,26 +23,88 @@ struct Identifier {
+    error NoExecutingDeposits();
+
+    /// @notice Thrown when trying to validate a cross chain message with a checksum
+    ///         that is invalid or was not provided in the transaction's access list to set the slot
+    ///         as warm.
+    error NotInAccessList();
+
     /// @notice Semantic version.
-    /// @custom:semver 1.0.0-beta.13
-    string public constant version = "1.0.0-beta.13";
+    /// @custom:semver 1.0.0-beta.14
+    string public constant version = "1.0.0-beta.14";
+
+    /// @notice The mask for the most significant bits of the checksum.
+    /// @dev    Used to set the most significant byte to zero.
+    bytes32 internal constant _MSB_MASK = 0x00ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
+
+    /// @notice Mask used to set the first byte of the bare checksum to 3 (0x03).
+    bytes32 internal constant _TYPE_3_MASK = 0x0300000000000000000000000000000000000000000000000000000000000000;
+
+    /// @notice The threshold to use to know whether the slot is warm or not.
+    uint256 internal constant _WARM_READ_THRESHOLD = 1000;
 
     /// @notice Emitted when a cross chain message is being executed.
     /// @param msgHash Hash of message payload being executed.
     /// @param id Encoded Identifier of the message.
     event ExecutingMessage(bytes32 indexed msgHash, Identifier id);
 
-    /// @notice Validates a cross chain message on the destination chain
-    ///         and emits an ExecutingMessage event. This function is useful
-    ///         for applications that understand the schema of the _message payload and want to
-    ///         process it in a custom way.
+    /// @notice Validates a cross chain message on the destination chain and emits an ExecutingMessage
+    ///         event. This function is useful for applications that understand the schema of the
+    ///         message payload and want to process it in a custom way.
+    /// @dev    Makes sure the checksum's slot is warm to ensure the tx included it in the access list.
     /// @param _id      Identifier of the message.
     /// @param _msgHash Hash of the message payload to call target with.
     function validateMessage(Identifier calldata _id, bytes32 _msgHash) external {
-        // We need to know if this is being called on a depositTx
-        if (IL1BlockInterop(Predeploys.L1_BLOCK_ATTRIBUTES).isDeposit()) revert NoExecutingDeposits();
+        bytes32 checksum = _calculateChecksum(_id, _msgHash);
+        (bool isWarm,) = _isWarm(checksum);
+        if (!isWarm) revert NotInAccessList();
 
         emit ExecutingMessage(_msgHash, _id);
     }
+
+    /// @notice Calculates a custom checksum for a cross chain message `Identifier` and `msgHash`.
+    /// @param _id The identifier of the message.
+    /// @param _msgHash The hash of the message.
+    /// @return checksum_ The checksum of the message.
+    function _calculateChecksum(Identifier memory _id, bytes32 _msgHash) internal pure returns (bytes32 checksum_) {
```

Comment:
For all of the 64-byte hashes in this function, it'd be a bit more efficient to just use `[0x00, 0x40)` rather than having to allocate net-new memory w/ `abi.encodePacked`. Not blocking though if you consider the readability hit concerning.

Reply:
Original spec PR had `yul` to do exactly that, but readability of `abi.encodePacked` won over that.